File: B53953/Laboratorio/Lab1/Orbis/src/main/java/com/orbis/b53953/domain/Autor.java
package com.orbis.b53953.domain;

public class Autor {

	private int idAutor;
	private String nombreAutor;
	private String apellidosAutor;
	
	
	public Autor(int idAutor, String nombreAutor, String apellidosAutor) {
		
		super();
		
		if(idAutor <= 100) throw new RuntimeException("El idAutor debe ser mayor a 100");
		if(nombreAutor == null || nombreAutor.trim().equals("")) throw new RuntimeException("El nombre del autor es requerido");
		if(apellidosAutor == null || apellidosAutor.trim().equals("")) throw new RuntimeException("Los apellidos del autor son requeridos");
		
		this.idAutor = idAutor;
		this.nombreAutor = nombreAutor;
		this.apellidosAutor = apellidosAutor;
	}


	public int getIdAutor() {
		return idAutor;
	}


	public void setIdAutor(int idAutor) {
		this.idAutor = idAutor;
	}


	public String getNombreAutor() {
		return nombreAutor;
	}


	public void setNombreAutor(String nombreAutor) {
		this.nombreAutor = nombreAutor;
	}


	public String getApellidosAutor() {
		return apellidosAutor;
	}


	public void setApellidosAutor(String apellidosAutor) {
		this.apellidosAutor = apellidosAutor;
	}


	@Override
	public String toString() {
		return "Autor [idAutor=" + idAutor + ", nombreAutor=" + nombreAutor + ", apellidosAutor=" + apellidosAutor
				+ "]";
	}
	
	
	
	
}

File: B67156/Lab´s/Laboratorio 4.1/OrbisB67156.Web/Orbis[B67156].Model/Domain/Autor.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Orbis_B67156_.Model.Domain
{
    class Autor
    {
        private int id_autor;
        public int Id_autor
        {
            get
            {
                return id_autor;
            }
            set
            {
                if (value <= 0) throw new Exception("Id autor es requerido");
                id_autor = value;
            }
        }
        private string nombre_autor;
        public string Nombre_autor
        {
            get
            {
                return nombre_autor;
            }
            set
            {
                if (string.IsNullOrEmpty(value)) throw new Exception("Nombre autor es requerido");
                nombre_autor = value;
            }
        }
        private string apellidos_autor;
        public string Apellidos_autor
        {
            get
            {
                return apellidos_autor;
            }
            set
            {
                if (string.IsNullOrEmpty(value)) throw new Exception("Apellidos autor es requerido");
                apellidos_autor = value;
            }
        }

        public Autor(int id_autor, string nombre_autor, string apellidos_autor)
        {
            Id_autor = id_autor;
            Nombre_autor = nombre_autor;
            Apellidos_autor = apellidos_autor;
        }

    }

}
